wordpress-mobile/MediaPicker-iOS
# Add badge to

This PR fixes part of [issue #9353](https://github.com/wordpress-mobile/WordPress-iOS/issues/9353), adding support for a custom badge at the top right corner of the picker cell.

The badge will show the file type.

## Configuration:

To configure what types show, a new option `badgedUTTypes` was introduced, with a list of file types that we want to show it's type in the badges.

## WPMediaAsset Conformance:

This also uses a new optional `WPMediaAsset` method `UTTypeIdentifier`, to get the file type information.

We could use the file extension, but using UTTypes seems to ensure the correct identification of the file.

We can also return `nil` if we don't want to show the badge of a filetype under specific conditions (like a gif with no animations)

## PHAsset issue:

The implementation of `UTTypeIdentifier` in the `PHAsset` was a bit tricky:
Using `[[PHAssetResource assetResourcesForAsset:self] firstObject] uniformTypeIdentifier];` was very slow, to the point that scrolling the picker gets choppy.

I tested using `[self valueForKey:@"uniformTypeIdentifier"];` and the performance is quite good. The downside of this is that it looks like it's using private api. I saw many people on StackOverflow using this, so we might not get rejected, but I'm not sure if we should take the risk.

## Style:
I opted to use the system blur effect, and adding a touch with vibrancy effect.
@iamthomasbishop could you please take a look?
PD: There is a small shadow but it's not visible in the screenshot for some reason :( 

![img_2861](https://user-images.githubusercontent.com/9772967/40726160-26b10a46-63f3-11e8-9720-ac67d2eb71bc.png)

To test:
- Save some GIFs images in your device image library.
- Run the app and see the badge appearing.
